### PR TITLE
YOLO network for ISCI Dataset - 46408855

### PR DESCRIPTION
**Pull request by LazyScribble**
# Overview
This project aimed to implement a YOLO network in order to detect skin lesions in the ISIC 2017 Dataset. This is task 3 on the task list. The provided code includes functionality for loading the dataset, as well as training, testing and predicting with the YOLO network.
# Files
* `dataset.py` : Contains the custom dataset for ISIC 2017 dataset. Will pre-process the images and create the bounding box attributes for the ground truth.
* `modules.py` : Contains the YOLO class, with the architecture and decoding function. Also contains the loss function class 'YOLO_loss', with helper function for calculating the IoU and filtering bounding boxes.
* `train.py` : When run, will train, test and save the model.
* `predict.py` : Uses the trained model to detect lesions in a given image and display the results.
# Results
This implementation of the YOLO network has failed to reach the desired IoU minimum for predictions which was 0.8. The average IoU for detections is currently 0.47.

### CODE REVIEW

**Comment by nathasha-naranpanawa** [comment]
**This is an initial inspection, no action is required at this point**

Difficulty: Normal

Readme:  OK
- no plots/images on classification results shown
- no training/testing loss or accuracy(mAP) plots shown (no proof of training)

Commit messages:  OK

Code:
- No hyperparameter tuning or attempts to improve performance
- minimal comments

Functionality/Performance:
- avergae IoU is below expected for test set (only 0.47)
- no classification accuracy shown

**Comment by shakes76** [comment]
# Marking
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation
Good spacing and comments
Header blocks missing -1
## Recognition Problem 
Maybe Solves problem, but no evidence of training -2
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage, no plots -1, no metrics of results from images -2, no classification accuracy -1
Module present
Commenting minimal -1
No Data leakage, test set? -1
Difficulty: Normal -5
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
ReadMe acceptable/good
Model/technical explanation
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
Feedback required, move readme into project folder, restore repo READMEs -2
Request Description good

**Comment by shakes76** [comment]
Feedback marks possible +2 if the requested changes are made (see above).

**Comment by wangzhaomxy** [comment]
Feedback attempt and feedback marks granted +2.
